yarn-project/circuits.js/src/abis/abis.ts

```diff
@@ -227,9 +229,19 @@ export async function computeContractAddress(
  * @param args - Arguments to hash.
  * @returns Pedersen hash of the arguments.
  */
-export async function computeVarArgsHash(wasm: CircuitsWasm, args: Fr[]): Promise<Fr> {
+export function computeVarArgsHash(wasm: CircuitsWasm, args: Fr[]): Promise<Fr> {
+  if (args.length === 0) return Promise.resolve(Fr.ZERO);
   wasm.call('pedersen__init');
-  return await abisComputeVarArgsHash(wasm, args);
+
+  const wasmComputeVarArgs = (args: Fr[]) =>
+    Fr.fromBuffer(wasmSyncCall(wasm, 'abis__compute_var_args_hash', new Vector(args), 32));
+
+  if (args.length > 32) {
+    const chunksHashes = chunk(args, 32).map(c => wasmComputeVarArgs(c));
+    return Promise.resolve(wasmComputeVarArgs(chunksHashes));
```

Comment:
This would misbehave in case there were more than 32^2 args. I know it's a bit of an absurd edge case but I would throw a nice error here if somebody pushed in more args just to be sure.